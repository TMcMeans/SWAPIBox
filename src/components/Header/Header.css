@font-face {
  font-family: 'star';
  font-style: normal;
  font-weight: normal;
  src: url(../../starjedi/Starjedi.ttf);
}

.header {
  display: flex;
  flex-direction: column;
  align-items: center;
  color: white;
  position: relative;
  z-index: 3;
}

.header-pic {
  opacity: .3;
  position: relative;
  z-index: 1;
  /* top: 145px; */
}

.main-title {
  font-size: 50px;
  letter-spacing: 10px;
  -webkit-transition: all 0.5s;
  -moz-transition: all 0.5s;
  transition: all 0.5s;
  color: #feda4a;
  font-family: 'star';
  -webkit-animation: neon2 1.5s ease-in-out infinite alternate;
  -moz-animation: neon2 1.5s ease-in-out infinite alternate;
  animation: neon2 1.5s ease-in-out infinite alternate;
  position: relative;
  z-index: 2;
  top: -140px;
}

@-webkit-keyframes neon2 {
  from {
    text-shadow: 0 0 10px #fff, 0 0 20px #fff, 0 0 30px #fff, 0 0 40px #feda4a, 0 0 70px #feda4a, 0 0 80px #feda4a, 0 0 100px #feda4a, 0 0 150px #feda4a;
  }
  to {
    text-shadow: 0 0 5px #fff, 0 0 10px #fff, 0 0 15px #fff, 0 0 20px #feda4a, 0 0 35px #feda4a, 0 0 40px #feda4a, 0 0 50px #feda4a, 0 0 75px #feda4a;
  }
}
