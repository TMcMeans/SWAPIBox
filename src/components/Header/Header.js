import React from 'react'


export const Header = () => {
  return (
    <div className="header">
        <h1>SWAPIBOX</h1>
        <button>Favorites</button>
      </div>
  )
}