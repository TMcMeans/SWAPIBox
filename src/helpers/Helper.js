export const getRandomFilm = filmArray => {
  const randomNumber = Math.floor(Math.random() * 6);
  return filmArray[randomNumber];
};

export const cleanPeopleData = async peopleArray => {
  const people = peopleArray.map(async person => {
    const { homeworld, species } = person;

    const fetches = [fetch(homeworld), fetch(species)];
    const response = await Promise.all(fetches);

    const parsedResponses = [response[0].json(), response[1].json()];

    const parsedData = await Promise.all(parsedResponses);

    const homeworldObj = parsedData[0];
    const speciesObj = parsedData[1];

    return {
      name: person.name,
      homeworld: homeworldObj.name,
      homePop: homeworldObj.population,
      species: speciesObj.name,
      isFavorite: false
    };
  });
  return Promise.all(people);
};

export const cleanVehicleData = async vehiclesArray => {
  const vehicles = await vehiclesArray.map(async vehicle => {
    return {
      name: vehicle.name,
      model: vehicle.model,
      class: vehicle.vehicle_class,
      passengers: vehicle.passengers,
      isFavorite: false
    };
  });
  return Promise.all(vehicles);
};

export const cleanPlanetsData = async planetsArray => {
  const planets = await planetsArray.map(async planet => {
    const { residents } = planet;
    const promisedPeople = residents.map(async resident => {
      const response = await fetch(resident);
      return response.json();
    });

    const resolvedResidents = await Promise.all(promisedPeople);

    return {
      name: planet.name,
      terrain: planet.terrain,
      population: planet.population,
      climate: planet.climate,
      residents: resolvedResidents,
      isFavorite: false
    };
  });
  return Promise.all(planets);
};
