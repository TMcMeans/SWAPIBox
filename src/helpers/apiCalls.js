export const fetchFilms = async () => {
  try {
    const url = 'https://swapi.co/api/films/';
    const response = await fetch(url);
    const films = await response.json();
    return films;
  } catch (err) {
    console.log(`Error: ${err.message}`);
  }
};

export const fetchPeople = async () => {
  try {
    const url = 'https://swapi.co/api/people/';
    const response = await fetch(url);
    const people = await response.json();
    return people;
  } catch (err) {
    console.log(`Error: ${err.message}`);
  }
};

export const fetchVehicles = async () => {
  try {
    const url = 'https://swapi.co/api/vehicles/';
    const response = await fetch(url);
    const vehicles = await response.json();
    return vehicles;
  } catch (err) {
    console.log(`Error: ${err.message}`);
  }
};

export const fetchPlanets = async () => {
  try {
    const url = 'https://swapi.co/api/planets/';
    const response = await fetch(url);
    const planets = await response.json();
    return planets;
  } catch (err) {
    console.log(`Error: ${err.message}`);
  }
};
