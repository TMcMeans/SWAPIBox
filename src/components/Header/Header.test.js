import React from 'react';
import { shallow } from 'enzyme';

import { Header } from './Header';
import { mockData } from '../../helpers/mockData';

describe('Header', () => {
  it('should match snapshot with all data passed in correctly', () => {
    const wrapper = shallow(<Header />);
    expect(wrapper).toMatchSnapshot();
  });
});
