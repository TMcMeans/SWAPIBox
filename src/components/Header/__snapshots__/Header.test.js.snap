// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should match snapshot with all data passed in correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Header />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="img-div"
        >
          <img
            className="header-pic"
            src="./vadarnew.png"
          />
        </div>,
        <h1
          className="main-title"
        >
          SWAPi Box
        </h1>,
      ],
      "className": "header",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <img
            className="header-pic"
            src="./vadarnew.png"
          />,
          "className": "img-div",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "header-pic",
            "src": "./vadarnew.png",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "SWAPi Box",
          "className": "main-title",
        },
        "ref": null,
        "rendered": "SWAPi Box",
        "type": "h1",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="img-div"
          >
            <img
              className="header-pic"
              src="./vadarnew.png"
            />
          </div>,
          <h1
            className="main-title"
          >
            SWAPi Box
          </h1>,
        ],
        "className": "header",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <img
              className="header-pic"
              src="./vadarnew.png"
            />,
            "className": "img-div",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "header-pic",
              "src": "./vadarnew.png",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "SWAPi Box",
            "className": "main-title",
          },
          "ref": null,
          "rendered": "SWAPi Box",
          "type": "h1",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
